### ISSUE TITLE
Tests leave key in keyring.

### ISSUE THREAD

**Issue by real-yfprojects** [borgbase/vorta#1215]
The issue was first brought up by @bastiencyr in #1065:

> Only for dev, I just noticed that there are a lot of keys added in my repo manager which start by "vorta-test-repo...". I don't know if it is a bug but it would be nice for dev to provide a way to remove this tests repo. I have something like 650 entries 'vorta-test-repo...' in my default password manager (gnome-keyring) :(

To solve this issue, the [keyring module of vorta](https://github.com/borgbase/vorta/blob/master/src/vorta/keyring/abc.py) needs a method to remove a key from the keyring. Currently I don't know of other uses of such a feature.

**Comment by ganeshrevadi** [comment]
Is this issue just refers to removing `key =lambda k: k[1]` from *get_keyring()* in vorta/src/vorta/keyring/abc.py 
@real-yfprojects

**Comment by real-yfprojects** [comment]
Nope, see above and also this comment: https://github.com/borgbase/vorta/pull/1584#issuecomment-1418987171

**Comment by real-yfprojects** [comment]
>  I understand that their needs to be a method `remove_password` added in the *abc.py* and its implementation in corresponding files.

Yes, start with defining the abstract method signature in `abc.py`. Then implement it for each password provider and adjust the abstract signature as needed. I suggest you start with secretstorage and do MacOS last.

When that works you can write test coverage for the new function and use it to clean up existing tests.

> Its a little complicated for me , if you could guide me through on how it work

No problem, I will do that. Just ask if run into an issue.

> what will be its usecases?

Currently we'll only use it for cleaning up tests.
